asterius/src/Asterius/Resolve.hs

```diff
@@ -108,18 +110,19 @@ linkStart ::
   [EntitySymbol] ->
   (AsteriusModule, Module, LinkReport)
 linkStart debug gc_sections verbose_err store root_syms export_funcs =
-  ( merged_m,
-    result_m,
-    mempty
-      { staticsSymbolMap = ss_sym_map,
-        functionSymbolMap = func_sym_map,
-        infoTableSet = makeInfoTableSet merged_m ss_sym_map,
-        Asterius.Types.LinkReport.tableSlots = tbl_slots,
-        staticMBlocks = static_mbs,
-        sptEntries = sptMap merged_m,
-        bundledFFIMarshalState = bundled_ffi_state
-      }
-  )
+  rnf merged_m
```

Comment:
The correct way to apply `rnf` is at `merged_m0`, as soon as gc-sections pass is over.

Reply:
Hmmm, given that everything is pure, I don't understand exactly what you mean by "as soon as [the] gc-sections pass is over". Do you mean based on the data dependencies? Something like this?
```diff
-   merged_m1
-     | debug = addMemoryTrap merged_m0
-     | otherwise = merged_m0
+   merged_m1
+     | rnf merged_m0 `seq` debug = addMemoryTrap merged_m0
+     | otherwise = merged_m0

```

Reply:
(well, basically `merged_m0 ``deepseq`` debug`)

Reply:
Introduce a `!merged_m0_evaluated = force merged_m0` binding between `merged_m0` and `merged_m1`, and use this in `merged_m1`.